packages/hydrogen/src/customer/customer.ts

```diff
@@ -174,7 +181,21 @@ export function createCustomerAccountClient({
     }
 
     try {
-      return parseJSON(body);
+      const APIresponse = parseJSON(body) as CustomerAPIResponse<T>;
+      const {errors} = APIresponse;
+
+      const gqlErrors = errors?.map(
+        ({message, ...rest}) =>
+          new GraphQLError(message, {
+            ...(rest as WritableDeep<typeof rest>),
+            clientOperation: `storefront.${errorOptions.type}`,
+            requestId: response.headers.get('x-request-id'),
+            queryVariables: variables,
+            query,
+          }),
+      );
+
+      return {...APIresponse, errors: gqlErrors};
```

Comment:
Minor change to avoid adding `errors: undefined` key to the result. Instead, this will omit the `errors` key. Just to make it behave similar to the Storefront client.

```suggestion
      return {...APIresponse, ...(errors && {errors: gqlErrors})};

```